### PR TITLE
DEPR: Panel4d and panelND

### LINKED ISSUE #13564: Deprecation of Panel4D/PanelND
Related to https://github.com/pydata/pandas/issues/13563 (deprecation of Panels), we could/should also deprecate Panel4D / PanelND. 
(xref https://github.com/pydata/pandas/issues/5049)

Deprecating Panel4D/ND could probably be done sooner than deprecating Panels (e.g. actually deprecating in 0.19.0 ?)

### PR DESCRIPTION
**Pull request by jreback**
closes #13564

### CODE REVIEW

**Comment by jreback** [comment]
This is was a PITA.

@jorisvandenbossche @wesm @shoyer

**Comment by jreback** [comment]
@shoyer @MaximilianR esp have a look at the warnings (e.g. maybe have a specific page)?

**Comment by shoyer** [comment]
Can we add the deprecation notice to the docstrings for `PanelND` and `Panel4D`, too?

@MaximilianR Let's see if we can finish up that doc page on xarray for panel users (https://github.com/pydata/xarray/pull/832) -- it would be great to be able to link to that in the pandas docs.

File: pandas/core/panel4d.py [hunk 1]
```diff
@@ -37,6 +38,9 @@
 def panel4d_init(self, data=None, labels=None, items=None, major_axis=None,
                  minor_axis=None, copy=False, dtype=None):
 
+    # deprecation GH13564
+    warnings.warn("Panel4D is deprecated and will be removed in a "
+                  "future version", FutureWarning, stacklevel=2)
```
Comment (jorisvandenbossche) [inline pandas/core/panel4d.py]: Maybe refer to xarray here as well? (like in the docs)

**Comment by jorisvandenbossche** [comment]
Oh boy, that will be a joy if when we can remove it :-) 
Too late for now, but isn't there a way to eg test the warning once, and be able to mute them for all other tests?

**Comment by jreback** [comment]
I wish - the problem is that FutureWarninh is generic so No easy way

**Comment by jorisvandenbossche** [comment]
Can't you filter with a regex on the message or module?

**Comment by jreback** [comment]
I gave it a try - u r welcome to change it - it's buggy in 27 I think

**Comment by jorisvandenbossche** [comment]
For the next big depr then :-)

**Comment by MaximilianR** [comment]
> @MaximilianR Let's see if we can finish up that doc page on xarray for panel users (pydata/xarray#832) -- it would be great to be able to link to that in the pandas docs.

I'll have a go this week!

**Comment by jreback** [comment]
@jorisvandenbossche updated.

@shoyer 
cc @MaximilianR 
we can always change the xarray link points later (I think we have an issue for that anyhow). As these need to go in the ``.to_xarray()`` doc-string as well.

File: pandas/api/tests/test_api.py [hunk 2]
```diff
@@ -58,11 +58,10 @@ class TestPDApi(Base, tm.TestCase):
 
     # these are already deprecated; awaiting removal
     deprecated_classes = ['SparsePanel', 'TimeSeries', 'WidePanel',
-                          'SparseTimeSeries']
+                          'SparseTimeSeries', 'Panel', 'Panel4D']
```
Comment (jorisvandenbossche) [inline pandas/api/tests/test_api.py]: I don't think `Panel` is already deprecated?

**Comment by jorisvandenbossche** [comment]
Small comment/question, for the rest ready to merge for me. The doc link can indeed be updated later.

**Pull request closed without merging**